frontend/src/components/login/login-input/LoginInput.tsx

```diff
@@ -69,12 +72,26 @@ const Login = () => {
   function resetButton() {
     setErrorMsg(null)
     setCanResend(false)
+    setOtp('')
   }
 
-  function resend() {
-    setOtpSent(false)
-    sendOtp()
+  async function resend() {
+    resetButton()
     sendUserEvent(GA_USER_EVENTS.RESEND_OTP)
+
+    try {
+      setIsResending(true)
+      await getOtpWithEmail(email)
+    } catch (err) {
+      setErrorMsg(err.message)
+      sendException(err.message)
+    } finally {
+      setIsResending(false)
+      // Always reset timer even if resending fails
+      setTimeout(() => {
+        setCanResend(true)
+      }, RESEND_WAIT_TIME)
+    }
```

Comment:
why not
```
    setIsResending(true)
    await sendOtp()
    setIsResending(false)
```

Reply:
and in catch block of `sendOtp()`
```
   setCanResend(true)
```

Reply:
I was thinking of enforcing the timeout even in the event that the resend failed (network error, etc). Not sure if that's the required behaviour though.